code/game/barricade.dm

```diff
@@ -0,0 +1,980 @@
+	return prob(max(30,(100.0*obj_integrity)/max_integrity))
+
+/obj/structure/barricade/examine(mob/user)
+	..()
+
+	if(is_wired)
+		to_chat(user, "<span class='info'>There is a length of wire strewn across the top of this barricade.</span>")
+	switch(damage_state)
+		if(0) to_chat(user, "<span class='info'>It appears to be in good shape.</span>")
+		if(1) to_chat(user, "<span class='warning'>It's slightly damaged, but still very functional.</span>")
+		if(2) to_chat(user, "<span class='warning'>It's quite beat up, but it's holding together.</span>")
+		if(3) to_chat(user, "<span class='warning'>It's crumbling apart, just a few more blows will tear it apart.</span>")
+
+/obj/structure/barricade/hitby(atom/movable/AM)
+	if(AM.throwing && is_wired)
+		if(iscarbon(AM))
+			var/mob/living/carbon/C = AM
+			C.visible_message("<span class='danger'>The barbed wire slices into [C]!</span>",
+			"<span class='danger'>The barbed wire slices into you!</span>")
+			C.apply_damage(10)
+			C.KnockDown(2) //Leaping into barbed wire is VERY bad
+	..()
+
+
+
+/obj/structure/barricade/Bumped(atom/A)
+	..()
+
+	if(istype(A, /mob/living/carbon/xenomorph/crusher))
+
+		var/mob/living/carbon/xenomorph/crusher/C = A
+
+		if(C.charge_speed < CHARGE_SPEED_MAX * 0.5)
+			return
+
+		if(crusher_resistant)
+			obj_integrity -= C.charge_speed * CRUSHER_CHARGE_BARRICADE_MULTI
+			update_health()
+
+		else if(!C.stat)
+			visible_message("<span class='danger'>[C] smashes through [src]!</span>")
+			destroy_structure()
+
+/obj/structure/barricade/CheckExit(atom/movable/O, turf/target)
+	if(closed)
+		return TRUE
+
+	if(O.throwing)
+		if(is_wired && iscarbon(O)) //Leaping mob against barbed wire fails
+			if(get_dir(loc, target) & dir)
+				return FALSE
+		return TRUE
+
+	if(get_dir(loc, target) & dir)
+		return FALSE
+	else
+		return TRUE
+
+/obj/structure/barricade/CanPass(atom/movable/mover, turf/target)
+	if(closed)
+		return TRUE
+
+	if(mover && mover.throwing)
+		if(is_wired && iscarbon(mover)) //Leaping mob against barbed wire fails
+			if(get_dir(loc, target) & dir)
+				return FALSE
+		return TRUE
+
+	if(istype(mover, /obj/vehicle))
+		var/obj/vehicle/roadtrafficaccident = mover
+		if(!roadtrafficaccident.demolish_on_ram)
+			return
+		visible_message("<span class='danger'>[mover] drives over and destroys [src]!</span>")
+		destroy_structure(0)
+		return FALSE
+
+	var/obj/structure/S = locate(/obj/structure) in get_turf(mover)
+	if(S && S.climbable && !(S.flags_atom & ON_BORDER) && climbable && isliving(mover)) //Climbable objects allow you to universally climb over others
+		return TRUE
+
+	if(get_dir(loc, target) & dir)
+		return FALSE
+	else
+		return TRUE
+
+/obj/structure/barricade/attack_animal(mob/user as mob)
+	return attack_alien(user)
+
+/obj/structure/barricade/attack_alien(mob/living/carbon/xenomorph/M)
+	M.do_attack_animation(src)
+	obj_integrity -= rand(M.xeno_caste.melee_damage_lower, M.xeno_caste.melee_damage_upper)
```

Comment:
obj_integrity -= rand(M.melee_damage_lower,M.melee_damage_upper)

Reply:
Oh yeah that'd make sense, no xenos

I think I have that now in change commit 746e6d4